src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -550,6 +550,48 @@ public IEnumerable<AccountBalance> GetBalances(string walletName, string account
             return balances;
         }
 
+        /// <inheritdoc />
+        public AddressBalance GetAddressBalance(string address)
+        {
+            Guard.NotEmpty(address, nameof(address));
+            this.logger.LogTrace("({0}:'{1}')", nameof(address), address);
+
+            AddressBalance balance = new AddressBalance
+            {
+                Address = address,
+                CoinType = this.coinType
+            };
+
+            lock (this.lockObject)
+            {
+                foreach (var wallet in this.Wallets)
+                {
+                    if (!wallet.ContainsAddress(new HdAddress { Address = address }))
+                        continue;
+
+
+                    HdAddress hdAddress = wallet.AccountsRoot
+                        .SelectMany(ac => ac.Accounts)
+                        .SelectMany(i => i.InternalAddresses)
+                        .SingleOrDefault(ad => ad.Address == address) ??
+                        wallet.AccountsRoot
+                            .SelectMany(ac => ac.Accounts)
+                            .SelectMany(e => e.ExternalAddresses)
+                            .SingleOrDefault(ad => ad.Address == address);
+
+                    if (hdAddress == null)
+                        continue;
+
+                    (Money AmountConfirmed, Money AmountUnconfirmed) result = hdAddress.GetSpendableAmount();
+
+                    balance.AmountConfirmed = result.AmountConfirmed;
+                    balance.AmountUnconfirmed = result.AmountUnconfirmed;
```

Comment:
Foreach will continue after the address is found. Please add `break` here to avoid that.